CRM/Extension/Mapper.php

```diff
@@ -318,6 +318,12 @@ public function getActiveModuleFiles($fresh = FALSE) {
         $this->cache->set($this->cacheKey . '_moduleFiles', $moduleExtensions);
       }
     }
+
+    // Since we're not caching the full path we add it now.
+    array_walk($moduleExtensions, function(&$value, $key) {
+      $value['filePath'] = $this->keyToPath($value['filePath']);
+    });
+
```

Comment:
Comments on  a few levels:

1. Agree that (all other things being equal) it's preferrable not to store absolute paths.
2. `r-code`: This refactoring has moved the code which throws `CRM_Extension_Exception`. Shouldn't the `try/catch` formulation also change?
3. `r-code`: It doesn't make sense for the field `filePath` to store the `full_name`. The field name should describe its content accurately. If you need to track a diff piece of data (*a `full_name` or `key`*), then it's OK to add a field called `full_name` or `key`.
4. Obviously, the main question with cutting out a cache is the performance impact.
    * I suppose some kind of benchmark approach would be best, though it might not be reasonable to do.
    * Considering it qualitatively...
        * *In support/defense of the change:* PHP logic runs faster now-a-days than it did when first written (PHP 5 vs 7). The extension-cache uses `prefetch`, so the whole ext-cache is pulled with one query. The extension-cache could potentially improve using `withArray=>fast`.
        * *In opposition/concern about the change*: I start feeling odd when a cache stores the output of a simple query. There can be multiple invocations of `getActiveModuleFiles()/getModuleExtensions()` in a single page-request, and each invocation would have to re-run the `keyToPath()` statements.

Reply:
2. Okay, I'll move the try/catch.
3. I was trying to keep it backwards compatible, though agree it's not accurate. Perhaps instead I'll add another element to the array `fullName` and leave `filePath` `NULL` until `keyToPath()` is run.
4. I guess my perspective was from a different angle. Currently can't cache this at all on some sites and need to maintain a patch, versus a bit of caching and no patch.

Perhaps we could have a static variable for the array to make it more efficient on a single page-request?

Reply:
2. I moved try/catch and tests are passing now. So that means 3. is probably also working well.

4. I may try the static variable now.

Reply:
@totten I think it's ready for another review. Tests are passing.

The try/catch is around the function which throws an Exception. I also made it a static variable to try to reduce the number of requests to `keyToPath()`. It'll only call it if the `filePath` is empty.